lib/IRGen/IRGenSIL.cpp

```diff
@@ -358,7 +358,10 @@ class IRGenSILFunction :
   /// Keeps track of the mapping of source variables to -O0 shadow copy allocas.
   llvm::SmallDenseMap<StackSlotKey, Address, 8> ShadowStackSlots;
   llvm::SmallDenseMap<Decl *, SmallString<4>, 8> AnonymousVariables;
-  llvm::SmallDenseMap<llvm::Instruction *, DominancePoint, 8> ValueVariables;
+  /// To avoid inserting elements into ValueDomPoints twice.
+  llvm::SmallDenseSet<llvm::Instruction *, 8> ValueVariables;
+  /// Holds the DominancePoint of values that are storage for a source variable.
+  SmallVector<std::pair<llvm::Instruction *, DominancePoint>, 8> ValueDomPoints;
```

Comment:
Hmm... it looks like the code that is necessary to make DominancePoint play nice with DenseMap is much more elaborate than just having the extra data structure here explicit. The simplest I could come up with is having DominancePoint inherit from void *.